### PR TITLE
V2

### PR DESCRIPTION
**Pull request by AlexGuler**

### CODE REVIEW

File: app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.gradle
```
Comment (lpalac4) [inline app/.gitignore:1]: replace all of those .idea with a single .idea/*

File: app/src/main/java/com/example/emojigame/Database/AppDatabase.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    entities = arrayOf(Puzzle::class)
+)
+abstract class AppDatabase : RoomDatabase() {
+
+    companion object {
+        private var INSTANCE: AppDatabase? = null
+
+        fun getInstance(context: Context): AppDatabase {
+            if (INSTANCE == null) {
+                synchronized(AppDatabase::class) {
```
Comment (lpalac4) [inline app/src/main/java/com/example/emojigame/Database/AppDatabase.kt:20]: #nth Being that you have no control in this class over how a client will use your abstract class - i would synchronize the method getInstance().  Say two threads hit getInstance() before initialization.  Thread A grabs lock, instantiates - Thread 2 grabs lock and does what?  You would have to check after grabbing lock that it hasnt been instantiated already.  So either sync on the method or add another check before instantiating

File: app/.gitignore (hunk 1 above)
Comment (AlexGuler) [reply app/.gitignore:1]: Ok!

File: app/src/main/java/com/example/emojigame/Database/AppDatabase.kt (hunk 2 above)
Comment (AlexGuler) [reply app/src/main/java/com/example/emojigame/Database/AppDatabase.kt:20]: Interesting. Thank you

**Pull request merged**